framework/utils/TreeStructure.py

```diff
@@ -321,30 +321,34 @@ def addChild(gNode,tNode,depth=0):
 ###########
 # PARSERS #
 ###########
-class XMLCommentParser(ET.XMLTreeBuilder):
-  """
-    An XML parser that expands on the default to preserves comments
-  """
-  def __init__(self):
-    """
-      Constructor.
-      @ In, None
-      @ Out, None
-    """
-    ET.XMLTreeBuilder.__init__(self)
-    self._parser.CommentHandler = self.handleComment
-
-  def handleComment(self,data):
-    """
-      Constructor.
-      @ In, data, string object to parse into comment
-      @ Out, None
-    """
-    self._target.start(ET.Comment,{})
-    self._target.data(data)
-    self._target.end(ET.Comment)
-
-#set up a parser for this module
+if sys.version_info.major > 2:
+  #In Python3, XMLParser is a C implementation, and can't be modified.
+  XMLCommentParser = ET.XMLParser
```

Comment:
I don't know that we had a test that explicitly tested retaining comments.  I think we've discussed before focusing on making our own comment tag instead of retaining the default XML comments?